docs/tilt.md

```diff
@@ -0,0 +1,30 @@
+# Rapid iterative development using Tilt
+
+## Requirements
+
+- [Tilt](https://tilt.dev/)
+- kubectl
+
+## Workflow
+
+Set your environment:
+
+```shell
+# Name of your cluster's k8s context (check kubectl config get-contexts).
+export IG_K8S_CLUSTER=my-cluster
+# A Docker registry to use for development. Tilt will push temporary images there.
+export IG_REGISTRY=quay.io
+# A Docker repository to use for development. Tilt will push temporary images there.
+export IG_REPOSITORY=me/inspektor-gadget
+```
+
+Ensure you are logged into your Docker registry (run `docker login` if necessary).
+
+Run Tilt:
+
+```shell
+tilt up
+```
+
+Hit Space and watch the progress in your browser. If you leave Tilt running in the background, any
+Tilt would rebuild, push and redeploy any code changes automatically when you modify source files.
```

Comment:
> when you modify source files

Which source files does it consider?
*.go and *.c?
Does it consider the .dockerignore files? (any changes in `cmd/*` is about the CLI so no need to rebuild the gadget container)

Reply:
I think it considers all files by default. I'm not sure how smart it is, i.e. maybe it triggers for doc changes, too.
Tilt does support ignoring files. Let me look into that and ensure we're filtering correctly.

Reply:
Also, I suspect we might need to trigger an eBPF rebuild in Tilt to ensure we bake up-to-date object files in the dev image.

Reply:
I personally have the (bad?) habit of hitting `Ctrl+S` many times to be sure I don't lose any change. Will it trigger the whole redeploy process each time I save a file?